server/aesdsocket.c

```diff
@@ -0,0 +1,210 @@
+	close(sockfd);
+	perror("Unable to bind Properly");
+	exit(-1);
+}
+syslog(LOG_USER, "Binded Connection from 9000");
+/*********************************************************************************** Perform the Daemonising Process *********************************************************************************/
+// The daemon process by accepting a statement -d as a command line argument to this file.
+/* Daemon Creation Process
+	Create the child process using fork
+	setsid() - set an Id and remove as a parent
+	change directory to /dev/null using chdir()
+	redirect to STDIN, STDOUT, and STDERR
+*/
+if(argc > 1){
+	if(strcmp(argv[1],"-d") == 0){
+		pid_t pid = fork();	//Create a child process using fork
+		if(pid < 0){
+			perror("Child Process not created: Daemon Process failed to create in step 1");
+			exit(-1);
+		}
+		else if(pid == 0){
+			if(setsid() < 0){
+				perror("Unable to set Id");
+				exit(-1);
+			}
+			
+			if(chdir("/") == -1){
+				perror("Unable to change directory");
+				exit(-1);
+			}
+			open("/dev/null",O_RDWR);
+			dup(0);
+			dup(0);
+			syslog(LOG_USER,"Daemon Created Successfully");
+		}
+		else{
+			exit(0);
+		}
+	}
+}
+/***************************************************************************************** Listening to the Server ***********************************************************************************/
+int listener = listen(sockfd, 10);
+if(listener < 0){											// Backlog value set as 10 as prescribed in the Beej's user guide on sockets.
+	perror("Failed to execute in the listening process");
+	return -1;
+}
+syslog(LOG_USER, "Listening phase completed");
+fd = open("/var/tmp/aesdsocketdata",O_RDWR|O_CREAT|O_APPEND, 0777);
+if(fd == -1){
+	perror("Unable to open the file");
+}
+syslog(LOG_USER,"File successfully opened");
+/*************************************************************************************************** Accepting packets *******************************************************************************/
+int bytes_received;
+int buffer_size = BUFFER_SIZE;
+struct sockaddr_in addr_client;
+socklen_t length;
+int newfd;
+int total_bytes_written_so_far = 0;
+while(1){
+	length = sizeof(addr_client);
+	newfd = accept(sockfd, (struct sockaddr *)&addr_client,&length);
+	if(newfd == -1){
+		perror("Unable to Accept connection from 9000");
+		exit(-1);
+	}
+	syslog(LOG_USER, "Accepted connection from 9000");
+	
+	bool packet_in_progress = true;
+	int space_occupied_buffer = 0;					// Store the total occupied bytes in memory
+	char *write_buffer = malloc(sizeof(char) * BUFFER_SIZE);		// allocate space from the heap memory
+
+        while(packet_in_progress)
+        {
+            bytes_received = recv(newfd, buffer, BUFFER_SIZE, 0);
+            if (bytes_received == 0)
+            {
+                packet_in_progress = false;
+            }
+            int bytes = 0;
+            while (bytes ++ < bytes_received)
+            {
+                if (buffer[bytes] == '\n')
```

Comment:
@srki3050 when `bytes++` increments to be == to `bytes_received` you will exceed the length of bytes received and possibly exceed array bounds here.  This is what the valgrind error is telling you.